src/software/ai/navigator/path_planner/path_planner_test.cpp

```diff
@@ -0,0 +1,239 @@
+                          {std::cos(3 * M_PI / 3), std::sin(3 * M_PI / 3)}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(4 * M_PI / 3), std::sin(4 * M_PI / 3)}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(5 * M_PI / 3), std::sin(5 * M_PI / 3)}),
+                      false),
+              },
+          .should_return_path = true},
+
+         {.name           = "Circle of robots surrounding ego at distance 0.2",
+          .start          = Point(0, 0),
+          .dest           = Point(4, 0),
+          .navigable_area = Rectangle({-5, -5}, {5, 5}),
+          .obstacles =
+              {
+                  Obstacle::createRobotObstacle(TestUtil::createRobotAtPos({0.2, 0}),
+                                                false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(M_PI / 3) * 0.2, std::sin(M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(2 * M_PI / 3) * 0.2, std::sin(2 * M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(3 * M_PI / 3) * 0.2, std::sin(3 * M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(4 * M_PI / 3) * 0.2, std::sin(4 * M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(5 * M_PI / 3) * 0.2, std::sin(5 * M_PI / 3) * 0.2}),
+                      false),
+              },
+          .should_return_path = false}};
+
+template <typename PlannerT>
+std::pair<std::string, PathPlannerConstructor> name_and_constructor()
+{
+    return std::pair<std::string, PathPlannerConstructor>(
+        typeid(PlannerT).name(), []() { return std::make_unique<PlannerT>(); });
+}
+
+std::vector<std::pair<std::string, PathPlannerConstructor>>
+    path_planner_names_and_constructors = {
+        // add path planner constructors here
+        name_and_constructor<ThetaStarPathPlanner>(),
+        // uncomment this if you want some tests to fail
+        //        name_and_constructor<StraightLinePathPlanner>()
+};
+
+
+void validatePath(const Path &path, const Point &start, const Point &dest,
+                  const Rectangle &navigable_area, const std::vector<Obstacle> &obstacles)
+{
+    for (long i = 0; i < std::lround(1.0 / PATH_CHECK_INTERVAL); i++)
```

Comment:
Can we avoid `PATH_CHECK_INTERVAL`, by making it proportional to the speed/min obstacle size or other parameters like that to avoid potentially skipping over an obstacle?

My only concern is that if we gave a start position at the top left corner of the field, and a  dest position at the bottom right corner, (which we probably won't do irl), we would skip over smaller obstacles because the parameterization from 0-1 is not being scaled by the time or even distance

Reply:
now we use a linear approximation to compute s per metre, and have a constant `PATH_CHECK_INTERVAL_M`